server/item/template_type.go

```diff
@@ -0,0 +1,133 @@
+package item
+
+type ArmourTrimTemplate struct {
```

Comment:
Is ArmourTrimTemplate an appropriate name if the netherite upgrade template can be used for tools as well? Also try and stick to dragonfly's enum convention [as seen here](https://github.com/df-mc/dragonfly/blob/master/server/item/banner_pattern_type.go)

Reply:
still not sure if I like the new name I provided in the newest commit lmk if you have any other idea for the name for this.